test/persistence-hooks.suite.js

```diff
@@ -575,6 +576,204 @@ module.exports = function(dataSource, should) {
           done();
         });
       });
+
+      describe('persist hook', function() {
+        it('triggers `persist` hook on create', function(done) {
+          TestModel.observe('persist', pushContextAndNext());
+
+          TestModel.create(
+            { id: 'new-id', name: 'a name' },
+            function(err, instance) {
+              if (err) return done(err);
+
+              observedContexts.should.eql(aTestModelCtx({
+                data: { id: 'new-id', name: 'a name' }
+              }));
+
+              done();
+            });
+        });
+
+        it('applies updates from `persist` hook - on create', function(done) {
+          TestModel.observe('persist', pushContextAndNext(function(ctx){
+            ctx.data.extra = 'hook data';
+          }));
+
+          TestModel.create(
+            { id: 'new-id', name: 'a name' },
+            function(err, instance) {
+              if (err) return done(err);
+              observedContexts.data.should.have.property('extra', 'hook data');
+              done();
+            });
+        });
+
+        it('triggers `persist` hook on save', function(done) {
+          TestModel.observe('persist', pushContextAndNext());
+
+          existingInstance.name = 'changed';
+          existingInstance.save(function(err, instance) {
+            if (err) return done(err);
+
+            observedContexts.should.eql(aTestModelCtx({
+              data: {
+                id: existingInstance.id,
+                name: 'changed'
+              },
+              currentInstance: {
+                id: existingInstance.id,
+                name: 'changed',
+                extra: undefined
+              },
+              options: { throws: false, validate: true }
+            }));
+
+            done();
+          });
+        });
+
+        it('applies updates from `persist` hook - on save', function(done) {
+          TestModel.observe('persist', pushContextAndNext(function(ctx){
+            ctx.data.extra = 'hook data';
+          }));
+
+          existingInstance.save(function(err, instance) {
+            if (err) return done(err);
+            observedContexts.data.should.have.property('extra', 'hook data');
```

Comment:
Same comment as above, please check that the created `instance` has the extra data set. You can look at "applies updates from `before save` hook" for inspiration.